docs/compressed-serialization.md

```diff
@@ -0,0 +1,196 @@
+# Compressed Serialization
+
+With the Chia hard fork at height 5'496'000, CLVM can be serialized in a more
+space efficient form, referring back to previous sub-trees instead of
+duplicating them. These references are referred to as "back references".
+
+## Format
+
+The original serialization format had 3 tokens.
+
+- `0xff` - a pair, followed by the left and right sub-trees.
+- `0x80` - NIL, an empty atom
+- Atom - serialized with a UTF-8 like encoding. (see [CLVM serialization](https://chialisp.com/clvm/#serialization))
```

Comment:
A property that clvm encoding has that UTF-8 encoding does not is all atoms serialize to a string that has the original atom as a substring. This property is actually used in python deserialization here https://github.com/richardkiss/clvm_rs/blob/main/wheel/python/clvm_rs/clvm_tree.py#L7

Reply:
The reason I point this out, besides being useful, is it shows that it's a little different from UTF-8, which sometimes mixes framing bits and content bits in the same bytes. CLVM serialization never does.